lightning/src/ln/channelmanager.rs

```diff
@@ -2651,6 +2651,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 							payee: payee.clone(),
 							final_value_msat: pending_amt_unsent,
 							final_cltv_expiry_delta: max_unsent_cltv_delta,
+							max_total_cltv_expiry_delta: DEFAULT_MAX_TOTAL_CLTV_DELTA,
```

Comment:
Should this be the value used in the `RouteParameters` when finding the `Route`? Similarly elsewhere?

Reply:
I'm not sure what the question is? Whether the default value of 1000 is always applicable?

Reply:
Yes, I stumbled also across that but refrained from too invasive changes. The issue is that the `RouteParameters` for the failed route are not available in `send_payment`/`send_payment_internal` etc. 
Maybe they should be  included in the `Route` struct and supplied fully to `get_route`? This way, every method calling `find_route`/`get_route` would have access to the parameters of the (failed) route?

Reply:
Yeah, I was thinking something like that. It would work here, but in the places where we form `PaymentPathFailed` events, we'll need it as part of `HTLCSource::OutboundRoute` or maybe have it directly in the path somehow? The latter is currently a `Vec` though. @TheBlueMatt Not sure if you have an opinion here. Seems it would need to be added to both `Route` and `HTLCSource::OutboundRoute`.

Reply:
Ah, hmmmm, I wonder if we should really move it back to `Payee`? Then we'd get this for free. It is kinda awkward in `RouteParameters` here, as we'd really want the payment retryer to use some user-configured value. Otherwise, yea, I think we'd want to put it in the `Route`.

Reply:
Hum, so while I see that putting it in `Payee` would be a quick fix, I'd think this could be pretty confusing, especially in the long term (since the limit itself can be either thought of to be a property of the route or the payer, but not the payee). 

So putting `RouteParameters` or directly `max_total_cltv_expiry_delta` in `Route` and `OutboundRoute` would make a bit more sense to me, even though it is also kind of redundant to have the same parameters in multiple structs. This is however already the case with these two structs and other variables.

An alternative could maybe be to supply the `max_total_cltv_expiry_delta` to `find_route`/`get_route` etc. and store the used value on a per-payment basis in `PendingOutboundPayment`?

Reply:
We could also take the view that `Payee` is simply different from what it says on the tin, and seek to rename it. The way the code is structured, absent any names, there's no question this should go in `Payee` - its used to communicate route parameters which can be used to calculate a new route to retry a payment. `RouteParameters` is info specific to the specific route. Really `final_cltv_expiry_delta` should be in the `Payee`, it only isn't because it can be trivially recalculated from the route itself.

I think I'd somewhat favor putting it in `Payee`, updating the docs, land this, and take renaming `Payee` as a followup. What do you think @jkczyz?

Reply:
> Hum, so while I see that putting it in `Payee` would be a quick fix, I'd think this could be pretty confusing, especially in the long term (since the limit itself can be either thought of to be a property of the route or the payer, but not the payee).
> 
> So putting `RouteParameters` or directly `max_total_cltv_expiry_delta` in `Route` and `OutboundRoute` would make a bit more sense to me, even though it is also kind of redundant to have the same parameters in multiple structs. This is however already the case with these two structs and other variables.
> 

I'd support putting `RouteParameters` in both `Route` and `OutboundRoute`, but for the latter I think we'd want to update the `final_value_msat` and `final_cltv_expiry_delta` values to reflect that of the path in question. Then we'd deprecate the `OutboundRoute::payee` field since it would be redundant with the one in `RouteParameters`.

Though, I guess this is a little redundant since `final_value_msat` and `final_cltv_expiry_delta` could be learned from the path.

> An alternative could maybe be to supply the `max_total_cltv_expiry_delta` to `find_route`/`get_route` etc. as a parameter and store the used value on a per-payment basis in `PendingOutboundPayment::Retryable`?

I guess this means the user would have to look it up in order to call `find_route` for a retry? We'd have to add a method for this to `ChannelManager` and `Payer` (trait in `lightning-invoice` crate). I'm more inclined to take the other approach, but could be convinced otherwise.

Reply:
> Yeah, it's starting to sink in that this may be more ideal. Would s/Payee/PaymentParameters and s/pubkey/payee_pubkey work for you?
> 
> We may event want to re-consider if RotueParameters is even necessary if final_cltv_expiry_delta is also moved. That just leaves the amount which can be recomputed from the path, IIUC.

Yea, both of those seem reasonable to me. I'm happy to drop the amount, too, but given its already there and a nice struct to contain "amount + parameters" I don't see why remove it.

Reply:
Alright, moving `max_total_cltv_expiry_delta` to `Payee` seems like a good solution, especially since renaming it to something along the line of `PaymentParameters` solves my issues with it. I now moved the value it with https://github.com/lightningdevkit/rust-lightning/pull/1234/commits/7fe7a763ad7e1df311b17bd991d596729ba928a4 and will open a follow-up PR for the renaming as soon as this is merged.